### PR TITLE
Fix deployment flow authentication state updates

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3080.

### CODE REVIEW

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 1]
```diff
@@ -33,7 +33,6 @@ class DeploymentFlow extends React.Component {
       deploying: false,
       credential: this.props.credential,
       loadingTerms: false,
-      loggedIn: this.props.isLoggedIn(),
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: If the logged in state is set here when the component is created then the `isLoggedIn()` method never gets called again to update the state.

**Comment by huwshimi** [comment]
QA:
- load the direct deploy flow when logged in (e.g. /new/?dd=cs:bundle/mediawiki-single-9)
- you should see the logged in version of the direct deploy
- log out and load the direct deploy, you should the logged out version
- click the login button and then you should get the logged in version

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7891/

**Comment by frankban** [comment]
QA is ok.
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**